### PR TITLE
Handle multi-word relations

### LINKED ISSUE #6: Multi-valued rel Attribute
Like "class" attribute in HTML elements, "rel" attribute of the link header can have multiple space separated values e.g., `rel="next last"` which I think will not be properly treated by the current code:

```js
var m = p.match(/\s*(.+)\s*=\s*"?([^"]+)"?/)
if (m) acc[m[1]] = m[2];
```

### PR DESCRIPTION
**Pull request by jarofghosts**
Fixes #6, multi-word `rel`s should be split into multiple objects.

### CODE REVIEW

**Comment by ibnesayeed** [comment]
Awesome! That was quick.

File: index.js [hunk 1]
```diff
@@ -34,7 +34,10 @@ module.exports = function (linkHeader) {
     .map(parseLink)
     .filter(function (x) { return x && x.rel; })
     .reduce(function (acc, x) {
-      acc[x.rel] = x;
+      x.rel.split(/\s+/).forEach(function (rel) {
+        acc[rel] = xtend(x, {rel: rel});
```
Comment (thlorenz) [inline index.js]: Please put some spacing around `{}`, i.e. `{ rel: rel }`.

File: index.js [hunk 2]
```diff
@@ -34,7 +34,10 @@ module.exports = function (linkHeader) {
     .map(parseLink)
     .filter(function (x) { return x && x.rel; })
     .reduce(function (acc, x) {
-      acc[x.rel] = x;
+      x.rel.split(/\s+/).forEach(function (rel) {
```
Comment (thlorenz) [inline index.js]: I know I'm guilty of this myself, but it'd make things faster to move this function declaration out of the `forEach` and name it.
Not crucial for this PR, but should consider doing this for all the others as well, i.e. `filter, reduce`. Basically the same way it's done for `map` above.

**Comment by thlorenz** [comment]
LGTM once spacing nit is fixed.

Please fix, squash into one commit and forward merge to master.
Thanks

File: index.js (hunk 2 above)
Comment (jarofghosts) [inline index.js]: totally agree, sorry was trying to fit surrounding style.

it looks like you prefer the function definitions at the top, then?

Comment (thlorenz) [inline index.js]: Yeah pull them to the highest scope possible, but as long as they aren't inside the `forEach` declaration it's all good.
However we don't have to worry about this right now, i.e. we can take care of this in another PR.

**Comment by jarofghosts** [comment]
merged in 1ad4bb81a2535d26a1f1cbe898e490c6ef797b3d

**Comment by thlorenz** [comment]
Thanks, published as 0.4.

When you got a chance to fix the anonymous functions I'll publish a patch (not urgent though).

**Comment by jarofghosts** [comment]
will probably PR it this weekend at some point.

thanks!

**Pull request closed without merging**